counter/server/lib/src/processes/counter.dart

```
@@ -0,0 +1,167 @@
+      name: 'ValidationService',
+      cmd: ValidateCounterName(name: event.name),
+      fac: CounterNameValidated.fromJson,
+    );
+
+    // if name is invalid finish the process with an error
+    if (validationResult.isValid) {
+      return FlowResult.error('counter name is invalid');
+    }
+
+    // create counter entity
+    await context.callEntity<CounterCreatedEvent>(
+      name: 'CounterEntity',
+      id: newCounterId,
+      cmd: CreateCounterCommand(
+        name: event.name,
+        initialValue: event.initialValue,
+      ),
+      fac: CounterCreatedEvent.fromJson,
+    );
+
+    // add the counter to the list
+    await context.callEntity<CounterAddedToListEvent>(
+      name: 'CounterListEntity',
+      id: kCounterListEntityId,
+      cmd: AddCounterToListCommand(counterId: newCounterId),
+      fac: CounterAddedToListEvent.fromJson,
+    );
+
+    return FlowResult.ok(newCounterId);
+  }
+
+  Future<FlowResult> delete(
+    DeleteCounterRequested event,
+    ProcessContext context,
+  ) async {
+    // delete entity (don't wait for a resulting event)
+    context.sendEntity(
+      name: 'CounterEntity',
+      id: event.counterId,
+      cmd: DeleteCounterCommand(),
+    );
+
+    // delete counter from the list (don't wait for a resulting event)
+    context.sendEntity(
+      name: 'CounterEntityEntity',
+      id: kCounterListEntityId,
+      cmd: RemoveCounterFromListCommand(counterId: event.counterId),
+    );
+
+    return FlowResult.ok();
+  }
+
+  Future<FlowResult> increment(
+    IncrementCounterRequested event,
+    ProcessContext context,
+  ) async {
+    try {
+      // increment counter (don't wait for a resulting event)
+      context.sendEntity(
+        name: 'CounterEntity',
+        id: event.counterId,
+        cmd: IncrementCounterCommand(amount: event.amount),
+      );
+      // TODO: do we throw entity defined excption type or just Exception
```

In Delurk we throw entity defined exception like `SpeechException`, etc. But this try/catch will not work here, because exceptions are ignored when using `send`.

---

@viacheslav-horda check if exception received on `call` is strongly typed

---

> [@viacheslav-horda](https://github.com/viacheslav-horda) check if exception received on `call` is strongly typed

FYI we actually throw a generic FluirFlowException, but it's message includes the original exception type.

```dart
    if (!env.isOk) {
      final err = FlowCallReplyErr.fromJson(env.reply);
      callInFlight.completer.completeError(
        FluirFlowException(
          'context.call() received error of type ${err.errorType} and message ${err.message}',
        ),
      );
      return;
    }
```

---

I've remove all try/catch statements